src/gurt/misc.c

```diff
@@ -11,12 +11,32 @@
 
 #include <stdarg.h>
 #include <math.h>
+#include <malloc.h>
 #include <string.h>
 #include <gurt/common.h>
 
 /* state buffer for DAOS rand and srand calls, NOT thread safe */
 static struct drand48_data randBuffer = {0};
 
+d_alloc_track_cb_t d_alloc_track_cb;
+d_alloc_track_cb_t d_free_track_cb;
+static __thread void *track_arg;
+
+void
+d_set_alloc_track_cb(d_alloc_track_cb_t alloc_cb, d_alloc_track_cb_t free_cb)
+{
+#ifndef DAOS_BUILD_RELEASE
+	d_alloc_track_cb = alloc_cb;
```

Comment:
do we have numbers on how much overhead this adds?  A lot of our benchmarking is done on 'dev' build.  Maybe we should make this debug by default?  Unless it's cheap.

Reply:
yeah, according to Dalton's test, this adds about 2-4% overhead.  So Johann requires to move this to dev build. 
Hmm,  the patch is needed to monitor the memory usage, and debug OOM issue IMHO.

>A lot of our benchmarking is done on 'dev' build. 

Hmm, we probably need be aware of that, if we really need some "good" number IMHO.